src/account.ts

```diff
@@ -144,20 +153,38 @@ export class Account {
     /**
      * @param receiverId NEAR account receiving the transaction
      * @param actions The transaction [Action as described in the spec](https://nomicon.io/RuntimeSpec/Actions.html).
+     * @param meta Free-form {@link TxMetadata}. If provided, whether it
+     *   requires a redirect through NEAR Wallet or not, the transaction's
+     *   outcome will be available in the {@link WalletAccount.completedTransactions} collection.
      * @returns {Promise<FinalExecutionOutcome>}
      */
-    protected async signAndSendTransaction(receiverId: string, actions: Action[]): Promise<FinalExecutionOutcome> {
+    protected async signAndSendTransaction(
+        receiverId: string,
+        actions: Action[],
+        meta: TxMetadata = null
+    ): Promise<FinalExecutionOutcome> {
         await this.ready;
 
-        let txHash, signedTx;
+        let txHash: Uint8Array, signedTx: SignedTransaction;
         // TODO: TX_NONCE (different constants for different uses of exponentialBackoff?)
         const result = await exponentialBackoff(TX_STATUS_RETRY_WAIT, TX_NONCE_RETRY_NUMBER, TX_STATUS_RETRY_WAIT_BACKOFF, async () => {
             [txHash, signedTx] = await this.signTransaction(receiverId, actions);
             const publicKey = signedTx.transaction.publicKey;
 
+            if (meta) {
+                cacheTransaction({
+                    hash: txHash.toString(),
```

Comment:
I wouldn't rely on `txHash.toString()` as `.toString()` is different between `Buffer` and `Uint8Array` which can be fun. I'd better convert explicitly like `Buffer.from(txHash).toString('hex')`